src/Our.Umbraco.Ditto/Attributes/UmbracoPropertyAttribute.cs

```diff
@@ -51,5 +52,10 @@ public UmbracoPropertyAttribute(string propertyName, string altPropertyName = ""
         /// Gets or sets the default value.
         /// </summary>
         public object DefaultValue { get; set; }
+
+        /// <summary>
+        /// Defines if property is an Archetype
+        /// </summary>
+        public bool IsArchetype { get; set; }
```

Comment:
:+1: That's where my thinking is at the moment - separate assembly to specifically to handle Archetype data.